opensrp-pnc/src/main/java/org/smartregister/pnc/repository/PncVisitInfoRepository.java

```diff
@@ -101,4 +117,188 @@ public boolean delete(Map<String, String> pncChild) {
         throw new NotImplementedException("");
     }
 
+    @NonNull
+    public List<Map<String, Object>> getPncVisitSummaries(@NonNull String motherBaseEntityId, int pageNo) {
+        List<Map<String, Object>> data = new ArrayList<>();
+
+        Cursor cursor = null;
+        Cursor subCursor = null;
+        try {
+            String[] visitIds = getVisitIds(motherBaseEntityId, pageNo);
+            String joinedIds = "'" + StringUtils.join(visitIds, "','") + "'";
+
+            String query = "SELECT * FROM " + Table.PNC_VISIT_INFO + " " +
+                    " WHERE " + PncVisitInfo.PARENT_BASE_ENTITY_ID + " ='" + motherBaseEntityId + "'  AND " + PncVisitInfo.BASE_ENTITY_ID + " IN (" + joinedIds + ") " +
+                    " ORDER BY " + PncVisitInfo.CREATED_AT + " DESC";
+
+            cursor = getReadableDatabase().rawQuery(query, null);
+
+            if (cursor.getCount() > 0) {
+                while (cursor.moveToNext()) {
+
+                    Map<String, Object> record = convert(cursor);
+
+                    List<Map<String, String>> childData = new ArrayList<>();
+                    record.put(CHILD_RECORDS, childData);
+
+                    String subQuery = "SELECT pb.first_name, pb.last_name, pb.dob, pvcs.* FROM pnc_visit_child_status AS pvcs " +
+                            "LEFT JOIN " + Table.PNC_BABY + " AS pb ON pb." + PncBaby.BASE_ENTITY_ID + " = pvcs." + PncVisitChildStatus.CHILD_RELATION_ID + " " +
+                            "WHERE pvcs." + PncVisitInfo.PARENT_BASE_ENTITY_ID + " = '" + record.get(PncVisitInfo.BASE_ENTITY_ID) + "'";
+
+                    subCursor = getReadableDatabase().rawQuery(subQuery, null);
+
+                    if (subCursor.getCount() > 0) {
```

Comment:
Could you clean this up ... like we talked about

Reply:
done